### PR TITLE
Router information and tags added to Sentry error logging

### PR DESCRIPTION
**Pull request by bobbykolev**
### Describe the background of your pull request
As part of the client's observability initiative, this PR aims to add additional information to our error logs.
https://github.com/alkem-io/client-web/issues/5843

### Changes
Route information was added on Sentry init. 
The Sentry error logs now support passing tags. 

I suggest keeping the tags (at least the keys) strictly defined, but more flexibility is possible.
The tags could be easily added to the other types of logs (warning, info, etc) but we don't use them ATM.

### CODE REVIEW

File: src/core/logging/sentry/log.ts [hunk 1]
```diff
@@ -8,14 +8,41 @@ const warningLevel: Sentry.SeverityLevel = 'warning';
...
+} as const;
+
+type TagsKeysType = keyof typeof tagKeys;
+type TagType = {
+  [key in TagsKeysType]?: string;
+};
+
+const setTags = (tags: TagType | undefined, scope: Sentry.Scope) => {
+  if (tags) {
+    for (const [key, value] of Object.entries(tags as TagType)) {
```
Comment (hero101) [inline src/core/logging/sentry/log.ts:30]: is type casting needed when we know **tags** is already defined?

File: src/core/logging/sentry/log.ts [hunk 2]
```diff
@@ -8,14 +8,41 @@ const warningLevel: Sentry.SeverityLevel = 'warning';
...
+  UI: 'UI',
+  WHITEBOARD: 'WHITEBOARD',
+} as const;
+
+type TagsKeysType = keyof typeof tagKeys;
+type TagType = {
+  [key in TagsKeysType]?: string;
+};
+
+const setTags = (tags: TagType | undefined, scope: Sentry.Scope) => {
```
Comment (hero101) [inline src/core/logging/sentry/log.ts:28]: i think it makes more sense the mandatory parameters to be first. Also it doesnt make sense to set tags when there are no tags provided.

**Review by hero101 — CHANGES_REQUESTED** [review]

**Review by hero101 — APPROVED** [review]

**Review by hero101 — APPROVED** [review]

File: src/core/logging/sentry/log.ts [hunk 3]
```diff
@@ -8,14 +8,39 @@ const warningLevel: Sentry.SeverityLevel = 'warning';
 const errorLevel: Sentry.SeverityLevel = 'error';
 const fatalLevel: Sentry.SeverityLevel = 'fatal';
 
+export const tagKeys = {
+  CATEGORY: 'CATEGORY',
+  LABEL: 'LABEL',
+} as const;
+
+export const tagCategoryValues = {
```
[the remark this answers is not in the document]
Comment (bobbykolev) [reply src/core/logging/sentry/log.ts:16]: Sure I'll switch to Enum in the next PR, thanks!

File: src/core/logging/sentry/log.ts [hunk 4]
```diff
@@ -8,14 +8,39 @@ const warningLevel: Sentry.SeverityLevel = 'warning';
 const errorLevel: Sentry.SeverityLevel = 'error';
 const fatalLevel: Sentry.SeverityLevel = 'fatal';
 
+export const tagKeys = {
+  CATEGORY: 'CATEGORY',
+  LABEL: 'LABEL',
```
[the remark this answers is not in the document]
Comment (bobbykolev) [reply src/core/logging/sentry/log.ts:13]: I prefer to keep them uppercase as constants and easy to spot as such on Sentry.

**Pull request merged**